DEPR update dedeprecation message for np.ptp  #28665 #6581

hasnain2808 opened the pull request:
- ref : https://github.com/pandas-dev/pandas/issues/28665#issuecomment-567499869 #6581
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

hasnain2808 committed:
update dedeprecation message for np.ptp DEPR #28665 #6581

jbrockmendel commented:
The fix here is not to update the warning but to remove Series.ptp entirely

hasnain2808 commented:
Here in #2865
https://github.com/pandas-dev/pandas/issues/28665#issuecomment-567499869
It was suggested that till the time this resolved we should update the message

gfyoung commented on pandas/core/generic.py:
```diff
@@ -10198,7 +10198,10 @@ def nanptp(values, axis=0, skipna=True):
                 "Method .ptp is deprecated and will be removed "
-                "in a future version. Use numpy.ptp instead.",
+                "in a future version. Use numpy.ptp instead."
+                "if you are already using numpy.ptp and still getting this message"
```
```suggestion
                "if you are already using numpy.ptp and still getting this message, "
```

hasnain2808 replied:
done

gfyoung commented on pandas/core/generic.py:
```diff
@@ -10198,7 +10198,10 @@ def nanptp(values, axis=0, skipna=True):
-                "in a future version. Use numpy.ptp instead.",
+                "in a future version. Use numpy.ptp instead."
+                "if you are already using numpy.ptp and still getting this message"
+                "please call to_numpy() to avoid this message in future calls"
```
```suggestion
                "please call to_numpy() to avoid this message in future calls. "
```

hasnain2808 replied:
done

gfyoung commented on pandas/core/generic.py:
```diff
@@ -10198,7 +10198,10 @@ def nanptp(values, axis=0, skipna=True):
+                "in a future version. Use numpy.ptp instead."
+                "if you are already using numpy.ptp and still getting this message"
+                "please call to_numpy() to avoid this message in future calls"
+                "eg : np.ptp(pd.Series([1, 2, 3]).to_numpy())",
```
```suggestion
                "For example: np.ptp(pd.Series([1, 2, 3]).to_numpy())",
```

hasnain2808 replied:
done

jreback reviewed (changes requested):
agree with @jbrockmendel let's just remove this entirely. it has been deprecated for a while and we are removing code for 1.0

hasnain2808 committed:
update dedeprecation message for np.ptp DEPR #28665 #6581 requested changes made

hasnain2808 commented:
> agree with @jbrockmendel let's just remove this entirely. it has been deprecated for a while and we are removing code for 1.0

so, for now, I have updated the message, 
I believe it will take some time for getting rid of np,ptp() till then this message should help.

jreback commented:
> > agree with @jbrockmendel let's just remove this entirely. it has been deprecated for a while and we are removing code for 1.0
> 
> so, for now, I have updated the message,
> I believe it will take some time for getting rid of np,ptp() till then this message should help.

no, for 1.0 we *are* going to remove this code, if you want to repurpose this PR to do it great.

hasnain2808 commented:
Ohhk no isssues
I would love to do it
As I am new to the codebase it would be really nice if you can give me some pointers around achieving this

jreback commented:
> Ohhk no isssues
> I would love to do it
> As I am new to the codebase it would be really nice if you can give me some pointers around achieving this

ok, you remove the code & associated tests, see https://dev.pandas.io/docs/whatsnew/v1.0.0.html#removal-of-prior-version-deprecations-changes for things that we *have* removed, so add an entry liek this

hasnain2808 committed:
remove code for ptp()

hasnain2808 commented:
so I removed code and tests for ptp
remove the part where series only functions are added
removed the entire series only function because ptp was the only series only function
deleted tests for pandas ptp function
but the test written for np.ptp gives me a future warning

jbrockmendel commented on pandas/core/generic.py:
```diff
@@ -10184,39 +10184,42 @@ def mad(self, axis=None, skipna=None, level=None):
-            ``numpy.ndarray`` method ``ptp``.""",
-            nanptp,
-        )
+    # @classmethod
```
delete all of this

hasnain2808 replied:
done

jbrockmendel commented on pandas/core/series.py:
```diff
@@ -4390,7 +4390,7 @@ def to_period(self, freq=None, copy=True):
 )
 Series._add_numeric_operations()
-Series._add_series_only_operations()
+# Series._add_series_only_operations()
```
delete.  basically never leave commented-out code in a PR

hasnain2808 replied:
actually made this commit to get some help on getting this future warning

hasnain2808 replied:
> so I removed code and tests for ptp
> remove the part where series only functions are added
> removed the entire series only function because ptp was the only series only function
> deleted tests for pandas ptp function
> but the test written for np.ptp gives me a future warning
> 
> `
> self = <pandas.tests.series.test_analytics.TestSeriesAnalytics object at 0x7f666e0267b8>
> 
> ```
> def test_ptp(self):
>     # GH21614
>     N = 1000
>     arr = np.random.randn(N)
>     ser = Series(arr)
>     with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
> ```
> 
> > ```
> >       assert np.ptp(ser) == np.ptp(arr)
> > ```
> 
> pandas/tests/series/test_analytics.py:861:
> `

jbrockmendel replied:
Remove the assert_produces_warning, since it shouldn't anymore.

hasnain2808 replied:
it works but can someone please explain to me the reason behind it so that I learn

jbrockmendel commented on pandas/tests/series/test_analytics.py:
```diff
@@ -861,37 +861,37 @@ def test_ptp(self):
+        # msg = r"Series\.ptp does not implement numeric_only\."
+        # with pytest.raises(NotImplementedError, match=msg):
+        #     with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
+        #         s.ptp(numeric_only=True)
```
delete

hasnain2808 replied:
done

hasnain2808 committed:
remove comments)

hasnain2808 committed:
remove comments

hasnain2808 committed:
solve errors

hasnain2808 commented:
tests are not failing at local no clue why azure pipelines are failing

`
(pandas-dev) moha@moha-HP-Pavilion-Notebook:~/projects/pandas-hasnain2808$ pytest pandas/tests/series/test_analytics.py
============================= test session starts ==============================
platform linux -- Python 3.6.9, pytest-5.3.2, py-1.8.0, pluggy-0.13.1
rootdir: /home/moha/projects/pandas-hasnain2808, inifile: setup.cfg
plugins: cov-2.8.1, forked-1.1.3, xdist-1.30.0, hypothesis-4.54.0
collected 168 items                                                            

pandas/tests/series/test_analytics.py ..........................s....... [ 20%]
........................................................................ [ 63%]
..............................................x.....x.....x...           [100%]

================== 164 passed, 1 skipped, 3 xfailed in 3.26s ===================
`

hasnain2808 committed:
test at which command azure pipelines failing

hasnain2808 committed:
use numpy.array_equal instead of ==

hasnain2808 committed:
use numpy.array_equal instead of ==

hasnain2808 committed:
Revert "use numpy.array_equal instead of =="

This reverts commit 5d8d35f9a918a5a37e6f035201b242887a407748.

hasnain2808 committed:
Revert "use numpy.array_equal instead of =="

This reverts commit 86c11c96cb32314a9487cc19e05d4b02b9615429.

hasnain2808 committed:
Revert "test at which command azure pipelines failing"

This reverts commit b28bc3ab43396dc4fe707637015d705b5482ada9.

hasnain2808 committed:
Revert "solve errors"

This reverts commit add3e23547b44b4628d722575889f9c14eec7996.

hasnain2808 committed:
Revert "remove comments)"

This reverts commit 08a5f3e90b9477b3e2d99601626760c5072d0095.

hasnain2808 committed:
Revert "remove comments"

This reverts commit 3df0283f072e10b9c1ab84334239a973a45f6c5a.

hasnain2808 committed:
Revert "update dedeprecation message for np.ptp DEPR #28665 #6581 requested changes made"

This reverts commit d89d7f657aba51ed90a4f04e31c1e30c4540ee31.

hasnain2808 committed:
Revert "remove code for ptp()"

This reverts commit d9e955e27ac6ae35d5c3423585323238cb9d11dd.

hasnain2808 committed:
removed ptp code and tests and also put a print statement to understand why azure build failing

jbrockmendel commented:
What was happening in the failing builds is that when we call `np.ptp(ser)` numpy first checks if `ser` has a `ptp` method, in which case it calls that.  This PR removes (removed) that method, so numpy is falling back to whatever it tries next, probably involving either `__array__` or `__array_func__` or something.  Whatever that something is, it's returning a Series when it should return a scalar.

hasnain2808 commented:
> What was happening in the failing builds is that when we call `np.ptp(ser)` numpy first checks if `ser` has a `ptp` method, in which case it calls that. This PR removes (removed) that method, so numpy is falling back to whatever it tries next, probably involving either `__array__` or `__array_func__` or something. Whatever that something is, it's returning a Series when it should return a scalar.

So I actually had tried what is numpy returning on my local machine. Over there it's returning float
And also it would be nice if you could guide me a bit more towards the solution of this numpy issue

hasnain2808 committed:
removed ptp code and tests and also put print statements to print type and values of ptp to understand why azure build failing

hasnain2808 commented:
> What was happening in the failing builds is that when we call `np.ptp(ser)` numpy first checks if `ser` has a `ptp` method, in which case it calls that. This PR removes (removed) that method, so numpy is falling back to whatever it tries next, probably involving either `__array__` or `__array_func__` or something. Whatever that something is, it's returning a Series when it should return a scalar.

As I am unable to reproduce this on my local machine I tried to push a commit with the print statements which are ignored by the pipeline.
Also only builds on some platforms are failing not all

hasnain2808 committed:
casting pandas to numpy before calling np.ptp()

hasnain2808 commented:
At last all tests passing
Requesting some reviewer to check it

jreback commented:
thanks for the effort @hasnain2808 I had already merged #30458 

happy to take a contribution on other issues.
